types/gh-pages/index.d.ts

```diff
@@ -3,8 +3,27 @@
 // Definitions by: Daniel Rosenwasser <https://github.com/DanielRosenwasser>
 //                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-
+export interface Git {
+    cwd: string;
+    cmd: string;
+    output: string;
+    exec: (command: string) => Promise<this | Error>;
+    init: () => Promise<this | Error>;
+    clean: () => Promise<this | Error>;
+    reset: (remote: string, branch: string) => Promise<this | Error>;
+    fetch: (remote: string) => Promise<this | Error>;
+    checkout: (remote: string, branch: string) => Promise<this | Error>;
+    rm: (files: string | string[]) => Promise<this | Error>;
+    add: (files: string | string[]) => Promise<this | Error>;
+    commit: (message: string) => Promise<this | Error>;
+    tag: (name: string) => Promise<this | Error>;
+    push: (remote: string, branch: string, force?: boolean) => Promise<this | Error>;
+    getRemoteUrl: (remote: string) => Promise<this | Error>;
+    deleteRef: (branch: string) => Promise<this | Error>;
+    clone: (repo: string, dir: string, branch: string, options: PublishOptions) => Promise<this | Error>;
+}
 export interface PublishOptions {
+    beforeAdd?: null | ((git: Git) => Promise<Git | Error>);
```

Comment:
```suggestion
    beforeAdd?: ((git: Git) => Promise<Git | void>) | null | undefined; 
```
this is nullable (I think it should support explicit null/undefined in strict props checks)
The return type can be void, it looks like 'git' instance is a local instance, not one returned by this custom method:
```js
.then(git => {
        return Promise.resolve(
          options.beforeAdd && options.beforeAdd(git)
        ).then(() => git);
      })
```
so the Readme example returns Git.rm (which is a promise), but that does not matter I think

Reply:
Using `undefined` instead of `void` per [build error](https://dev.azure.com/definitelytyped/DefinitelyTyped/_build/results?buildId=122452&view=logs&j=8635df7d-ecde-52a7-c7a8-84b998e35690&t=c264d665-0b79-59b7-7e04-9965cbe74860).

```
ERROR: 23:46  void-return  Use the `void` type for return types only. Otherwise, use `undefined`.
```